src/general/ModLoader.cs

```diff
@@ -0,0 +1,320 @@
+
+            var currentModInfo =
+                JsonConvert.DeserializeObject<ModInfo>(ReadJSONFile(currentMod.FullName + "/mod_info.json"));
+
+            if (currentModInfo.AutoLoad)
+            {
+                continue;
+            }
+
+            currentModInfo.Location = currentMod.FullName;
+            modList.Add(currentModInfo);
+            unloadedItemList.AddItem(currentModInfo.ModName);
+        }
+    }
+
+    private void OnLoadPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (loadedModList.Count <= 0)
+        {
+            return;
+        }
+
+        foreach (ModInfo currentMod in loadedModList)
+        {
+            if (string.IsNullOrEmpty(currentMod.Dll))
+            {
+                if (File.Exists(currentMod.Location + "/" + currentMod.Dll))
+                {
+                    Assembly.LoadFile(currentMod.Location + "/" + currentMod.Dll);
+                }
+            }
+
+            if (!File.Exists(currentMod.Location + "/mod.pck"))
+            {
+                GD.Print("Fail to find mod file: " + currentMod.ModName);
+                continue;
+            }
+
+            if (ProjectSettings.LoadResourcePack(currentMod.Location + "/mod.pck", true))
+            {
+                GD.Print("Loaded mod: " + currentMod.ModName);
+            }
+            else
+            {
+                GD.Print("Failed to load mod: " + currentMod.ModName);
+            }
+        }
+
+        GD.Print("All mods loaded");
+        SceneManager.Instance.ReturnToMenu();
+    }
+
+    private void OnBackPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        EmitSignal(nameof(OnModLoaderClosed));
+    }
+
+    private void OnResetPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        confirmationPopup.PopupCentered();
+    }
+
+    private void ResetGame()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (!File.Exists(Directory.GetCurrentDirectory() + "/Thrive.pck"))
+        {
+            GD.Print("Fail to find Thrive");
+            return;
+        }
+
+        if (ProjectSettings.LoadResourcePack(Directory.GetCurrentDirectory() + "/Thrive.pck", true))
+        {
+            GD.Print("Reset successful");
+        }
+        else
+        {
+            GD.Print("Reset failed");
+        }
+
+        SceneManager.Instance.ReturnToMenu();
+    }
+
+    private void OnMoveUpPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (loadedItemList.GetSelectedItems().Length > 0)
+        {
+            var selectedItemIndex = loadedItemList.GetSelectedItems()[0];
+            if (selectedItemIndex == 0)
+            {
+                return;
+            }
+
+            loadedItemList.MoveItem(selectedItemIndex, selectedItemIndex - 1);
+            selectedItemIndex = loadedItemList.GetSelectedItems()[0];
+            MoveItem(loadedModList, selectedItemIndex + 1, selectedItemIndex);
```

Comment:
Would it be possible to just run the MoveItems in all cases? It seems that some godot child reordering code just calls the move to put things in the right order even if they already are.
This would reduce a lot of similar code.